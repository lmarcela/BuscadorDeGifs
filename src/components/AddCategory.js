import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) =>{   
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) =>{   
        e.preventDefault(); 
        if(inputValue.trim().length>2){
            setCategories( c => [inputValue,...c]);
            setInputValue('');
        }             
    }
    return (
          <form className="w3-container" onSubmit={handleSubmit}>
            <input className="w3-input w3-border" type="text" value={inputValue} onChange={handleInputChange} id="in0"/>  
          </form>
    )
}
    
AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
