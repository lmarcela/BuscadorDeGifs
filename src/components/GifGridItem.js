import React from 'react'
import PropTypes from 'prop-types'
export const GifGridItem = ({title, url}) => {
    return (
        <>        
            <div className="w3-third">
                <div className="w3-card w3-center animate__animated animate__slideInUp animate__faster">
                    <img src={url} alt={title} style={{height:200, maxWidth:"-webkit-fill-available"}} />
                    <p>{title?.trim() || "Sin titulo"}</p>
                </div>
            </div>
        </>
    )
}
GifGridItem.propTypes={
    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
