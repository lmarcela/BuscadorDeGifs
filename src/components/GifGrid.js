import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {
    const {data:imagenes, loading} = useFetchGifs(category);
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
            
            {<div className="w3-row-padding">
                {
                    imagenes.map((img) => (
                        <GifGridItem key={img.id} {...img}/>
                    )) 
                }
            </div>}
            
            {!loading && imagenes.length===0 && <p>Sin resultados</p>}
        </>
    )
}
