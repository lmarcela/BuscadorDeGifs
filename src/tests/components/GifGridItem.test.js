import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />',()=>{
    const title = "Prueba de Titulo";
    const url = "https://imagen.jpg";
    
    const wrapper=shallow(<GifGridItem title={title} url={url}/>);   

    test('debe mostrar <GifGridItem /> correctamente', () => {     
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => { 
        const p = wrapper.find('p');    
        expect(p.text().trim()).toBe(title);
    })

    test('debe de tener la imagen igual al url y alt de los props', () => { 
        const imagen = wrapper.find('img');    
        //console.log(imagen.props())
        expect(imagen.prop('src')).toBe(url);
        expect(imagen.prop('alt')).toBe(title);
    })

    test('debe de tener animate__slideInUp', () => { 
        const div = wrapper.find('div').at(1);    
        expect(div.prop('className').includes("animate__slideInUp")).toBe(true);
    })
});


