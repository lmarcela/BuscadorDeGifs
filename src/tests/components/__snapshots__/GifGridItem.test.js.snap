// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<Fragment>
  <div
    className="w3-third"
  >
    <div
      className="w3-card w3-center animate__animated animate__slideInUp animate__faster"
    >
      <img
        alt="Prueba de Titulo"
        src="https://imagen.jpg"
        style={
          Object {
            "height": 200,
            "maxWidth": "-webkit-fill-available",
          }
        }
      />
      <p>
        Prueba de Titulo
      </p>
    </div>
  </div>
</Fragment>
`;
